python-frontend/src/main/java/org/sonar/python/tree/PythonTreeMaker.java

```diff
@@ -1634,7 +1634,26 @@ private FormatSpecifier formatSpecifier(AstNode expressionNode) {
   }
 
   private FormattedExpression formattedExpression(AstNode expressionNode) {
-    Expression exp = expression(expressionNode.getFirstChild(PythonGrammar.TEST));
+    var formattedExpressionEntities = expressionNode.getFirstChild(PythonGrammar.FORMATTED_EXPR_ENTITIES);
+    var expressions = formattedExpressionEntities
+      .getChildren(PythonGrammar.TEST)
+      .stream()
+      .map(this::expression)
+      .collect(Collectors.toList());
+
+    Expression exp;
+    if (expressions.size() == 1) {
+      exp = expressions.get(0);
+    } else {
+      var commas = formattedExpressionEntities
+        .getChildren(PythonPunctuator.COMMA)
+        .stream()
+        .map(AstNode::getToken)
+        .map(PythonTreeMaker::toPyToken)
+        .collect(Collectors.toList());
+
+      exp = new ExpressionListImpl(expressions, commas);
```

Comment:
Official AST from Python yields a Tuple instead of an expression list:

```
>>> print(ast.dump(ast.parse("f'{42,x}'"), indent=4))
Module(
    body=[
        Expr(
            value=JoinedStr(
                values=[
                    FormattedValue(
                        value=Tuple(
                            elts=[
                                Constant(value=42),
                                Name(id='x', ctx=Load())],
                            ctx=Load()),
                        conversion=-1)]))],
    type_ignores=[])
```

Reply:
done